src/traces/treemap/attributes.js

```diff
@@ -0,0 +1,273 @@
+
+        pad: {
+            valType: 'number',
+            role: 'style',
+            min: 0,
+            dflt: 3,
+            editType: 'plot',
+            description: [
+                'Sets the inner padding (in px).'
+            ].join(' ')
+        },
+
+        editType: 'calc',
+    },
+
+    marker: extendFlat({
+        pad: {
+            t: {
+                valType: 'number',
+                role: 'style',
+                min: 0,
+                editType: 'plot',
+                description: [
+                    'Sets the padding form the top (in px).'
+                ].join(' ')
+            },
+            l: {
+                valType: 'number',
+                role: 'style',
+                min: 0,
+                editType: 'plot',
+                description: [
+                    'Sets the padding form the left (in px).'
+                ].join(' ')
+            },
+            r: {
+                valType: 'number',
+                role: 'style',
+                min: 0,
+                editType: 'plot',
+                description: [
+                    'Sets the padding form the right (in px).'
+                ].join(' ')
+            },
+            b: {
+                valType: 'number',
+                role: 'style',
+                min: 0,
+                editType: 'plot',
+                description: [
+                    'Sets the padding form the bottom (in px).'
+                ].join(' ')
+            },
+
+            editType: 'calc'
+        },
+
+        colors: sunburstAttrs.marker.colors,
+
+        opacity: {
+            valType: 'number',
+            editType: 'style',
+            role: 'style',
+            min: 0,
+            max: 1,
+            description: [
+                'Sets the opacity of the branches i.e. based on the hierarchy height.',
+                'With colorscale it is defaulted to 1; otherwise it is defaulted to 0.7'
+            ].join(' ')
+        },
+
+        line: sunburstAttrs.marker.line,
+
+        editType: 'calc'
+    },
+        colorScaleAttrs('marker', {
+            colorAttr: 'colors',
+            anim: false // TODO: set to anim: true?
+        })
+    ),
+
+    leaf: {
+        opacity: {
+            valType: 'number',
+            editType: 'style',
+            role: 'style',
+            min: 0,
+            max: 1,
+            dflt: 1,
+            description: [
+                'Sets the opacity of the leaves.'
+            ].join(' ')
+        },
+        editType: 'plot'
+    },
+
+    pathbar: {
+        visible: {
+            valType: 'boolean',
+            dflt: true,
+            role: 'info',
+            editType: 'plot',
+            description: [
+                'Determines if the path bar is drawn.'
+            ].join(' ')
+        },
+
+        side: {
+            valType: 'enumerated',
+            values: [
+                'top',
+                'bottom'
+            ],
+            dflt: 'top',
```

Comment:
IMO - the `pathbar.side` should be defaulted to `bottom`  i.e. to avoid mixing with headers especially when the entry point is the level is 1 (at a child of the root).

Reply:
cc https://github.com/plotly/plotly.js/pull/4185#discussion_r323917152

Reply:
Let's keep the pathbar at the top by default. Resolved.